tools/esdlcomp/esdlcomp.cpp

```diff
@@ -1154,6 +1152,61 @@ ESDLcompiler::ESDLcompiler(const char * sourceFile, bool generatefile, const cha
     }
 }
 
+bool ESDLcompiler::locateIncludedFile(StringBuffer& filepath, const char* prot, const char* srcDir, const char* fname, const char* ext)
+{
+    StringBuffer alternateExtFilename;
+    alternateExtFilename.setf("%s%s%s", (prot && *prot) ? prot : "", srcDir, fname);
+
+    const char* alt_ext;
+    if (stricmp(ext, LEGACY_FILE_EXTENSION)==0)
+        alt_ext = ESDL_FILE_EXTENSION;
+    else
+        alt_ext = LEGACY_FILE_EXTENSION;
+    alternateExtFilename.append(alt_ext);
+
+    OwnedIFile fileInSrcDir = createIFile(alternateExtFilename.str());
+    if (fileInSrcDir->exists())
+    {
+        filepath.set(alternateExtFilename.str());
+        return true;
+    }
+
+    if (includePath)
+    {
+        StringArray includeDirs;
+        includeDirs.appendList(includePath, ENVSEPSTR);
+        ForEachItemIn(x, includeDirs)
+        {
+            const char* dir = includeDirs.item(x);
+            if (dir && *dir)
+            {
+                StringBuffer pathInInclude(dir);
+                pathInInclude.trim();
+                if (pathInInclude.charAt(pathInInclude.length() - 1) != PATHSEPCHAR)
+                    pathInInclude.append(PATHSEPCHAR);
+                pathInInclude.append(fname);
+                VStringBuffer pathInIncludeFull("%s%s", pathInInclude.str(), ext);
+                OwnedIFile fileInInclude = createIFile(pathInIncludeFull.str());
+                if (fileInInclude->exists())
+                {
+                    filepath.set(pathInIncludeFull.str());
+                    return true;
+                }
+                VStringBuffer altPathInIncludeFull("%s%s", pathInInclude.str(), alt_ext);
```

Comment:
You could reuse the existing variable and do 

pathInIncludeFull.setf("%s%s", pathInInclude.str(), alt_ext);

here.